BUG: Regression in fonts for 1.5.16

On 1.5.16 I get a weird font error on [Azure](https://dev.azure.com/LABSN/expyfun/_build/results?buildId=251&view=logs&j=09f0a5bc-9cbc-582e-eae5-2a9f4ac9c7a3&t=a18c87f3-e3b1-5637-4658-5c64116ee71b&l=1285) (and AppVeyor) on Windows:
```
# First line is initializing a pyglet.text.layout.TextLayout from a doc=pyglet.text.decode_attributed
...
expyfun\visual\_visual.py:116: in __init__
    dpi=int(ec.dpi))
c:\hostedtoolcache\windows\python\3.7.9\x64\lib\site-packages\pyglet\text\layout.py:866: in __init__
    self.document = document
c:\hostedtoolcache\windows\python\3.7.9\x64\lib\site-packages\pyglet\text\layout.py:895: in document
    self._init_document()
c:\hostedtoolcache\windows\python\3.7.9\x64\lib\site-packages\pyglet\text\layout.py:1300: in _init_document
    self._update()
c:\hostedtoolcache\windows\python\3.7.9\x64\lib\site-packages\pyglet\text\layout.py:1211: in _update
    lines = self._get_lines()
c:\hostedtoolcache\windows\python\3.7.9\x64\lib\site-packages\pyglet\text\layout.py:1188: in _get_lines
    glyphs = self._get_glyphs()
c:\hostedtoolcache\windows\python\3.7.9\x64\lib\site-packages\pyglet\text\layout.py:1338: in _get_glyphs
    for start, end, (font, element) in runs.ranges(0, len(text)):
c:\hostedtoolcache\windows\python\3.7.9\x64\lib\site-packages\pyglet\text\runlist.py:410: in ranges
    starts, ends, values = zip(*[next(i) for i in iterators])
c:\hostedtoolcache\windows\python\3.7.9\x64\lib\site-packages\pyglet\text\runlist.py:410: in <listcomp>
    starts, ends, values = zip(*[next(i) for i in iterators])
c:\hostedtoolcache\windows\python\3.7.9\x64\lib\site-packages\pyglet\text\document.py:720: in ranges
    ft = font.load(font_name, font_size, bold=bold, italic=italic, stretch=stretch, dpi=self.dpi)
c:\hostedtoolcache\windows\python\3.7.9\x64\lib\site-packages\pyglet\font\__init__.py:144: in load
    font = _font_class(name, size, bold=bold, italic=italic, stretch=stretch, dpi=dpi)
c:\hostedtoolcache\windows\python\3.7.9\x64\lib\site-packages\pyglet\font\win32.py:495: in __init__
    super(GDIPlusFont, self).__init__(name, size, bold, italic, stretch, dpi)
c:\hostedtoolcache\windows\python\3.7.9\x64\lib\site-packages\pyglet\font\win32.py:229: in __init__
    self.logfont = self.get_logfont(name, size, bold, italic, dpi)
c:\hostedtoolcache\windows\python\3.7.9\x64\lib\site-packages\pyglet\font\win32.py:260: in get_logfont
    logfont.lfItalic = italic
E   TypeError: an integer is required (got type NoneType)
```
On 1.5.15 [there is no error](https://dev.azure.com/LABSN/expyfun/_build/results?buildId=252&view=logs&j=09f0a5bc-9cbc-582e-eae5-2a9f4ac9c7a3&t=a18c87f3-e3b1-5637-4658-5c64116ee71b). Any ideas what might be the cause and/or solution? Did anything change in terms of win32 fonts or TextLayout between releases?

--- a/pyglet/text/document.py
+++ b/pyglet/text/document.py
@@ -40,7 +40,7 @@ Abstract representation
 
 Styled text in pyglet is represented by one of the :py:class:`~pyglet.text.document.AbstractDocument` classes,
 which manage the state representation of text and style independently of how
-it is loaded or rendered.  
+it is loaded or rendered.
 
 A document consists of the document text (a Unicode string) and a set of
 named style ranges.  For example, consider the following (artificial)
@@ -131,7 +131,7 @@ entire paragraph, otherwise results are undefined.
 ``align``
     ``left`` (default), ``center`` or ``right``.
 ``indent``
-    Additional horizontal space to insert before the first 
+    Additional horizontal space to insert before the first
 ``leading``
     Additional space to insert between consecutive lines within a paragraph,
     in points.  Defaults to 0.
@@ -159,7 +159,7 @@ document; it will be ignored by the built-in text classes.
 
 All style attributes (including those not present in a document) default to
 ``None`` (including the so-called "boolean" styles listed above).  The meaning
-of a ``None`` style is style- and application-dependent. 
+of a ``None`` style is style- and application-dependent.
 
 .. versionadded:: 1.1
 """
@@ -181,8 +181,8 @@ class InlineElement:
 
     Elements behave like a single glyph in the document.  They are
     measured by their horizontal advance, ascent above the baseline, and
-    descent below the baseline.  
-    
+    descent below the baseline.
+
     The pyglet layout classes reserve space in the layout for elements and
     call the element's methods to ensure they are rendered at the
     appropriate position.
@@ -225,8 +225,8 @@ class InlineElement:
 
         It is the responsibility of the element to clip itself against
         the layout boundaries, and position itself appropriately with respect
-        to the layout's position and viewport offset.  
-        
+        to the layout's position and viewport offset.
+
         The `TextLayout.top_state` graphics state implements this transform
         and clipping into window space.
 
@@ -263,7 +263,7 @@ class AbstractDocument(event.EventDispatcher):
     This class can be overridden to interface pyglet with a third-party
     document format.  It may be easier to implement the document format in
     terms of one of the supplied concrete classes :py:class:`~pyglet.text.document.FormattedDocument` or
-    :py:class:`~pyglet.text.document.UnformattedDocument`. 
+    :py:class:`~pyglet.text.document.UnformattedDocument`.
     """
     _previous_paragraph_re = re.compile(u'\n[^\n\u2029]*$')
     _next_paragraph_re = re.compile(u'[\n\u2029]')
@@ -723,7 +723,7 @@ class _FontStyleRunsRangeIterator:
     def __getitem__(self, index):
         from pyglet import font
         font_name, font_size, bold, italic, stretch = self.zip_iter[index]
-        return font.load(font_name, font_size, bold=bold, italic=italic, stretch=stretch, dpi=self.dpi)
+        return font.load(font_name, font_size, bold=bool(bold), italic=bool(italic), stretch=stretch, dpi=self.dpi)
 
 
 class _NoStyleRangeIterator:

